cli/azd/internal/appdetect/spring_boot.go

```diff
@@ -90,17 +90,21 @@ var databaseDependencyRules = []DatabaseDependencyRule{
 	},
 }
 
+// todo: remove parentProject, when passed in the mavenProject is the effective pom.
```

Comment:
I plan to do this in the next PR.

Reply:
> when passed in, the mavenProject is the effective pom.

No matter the effect pom is analyzed by `mvn help:effective-pom` or analyze by our go lang code.